File: src/euler/triangular_12.py
#https://projecteuler.net/problem=12
import math
from sympy.ntheory import factorint

def solve():
    i = 2
    tri_num = 1
    while num_factors(tri_num) < 500:
        i += 1
        tri_num += i
    print("{} has +500 factors".format(tri_num))

def num_factors(n):
    prime_fact = factorint(n)
    factors = 1
    for k in prime_fact.keys():
        factors *= prime_fact[k]+1
    return factors

File: src/euler/triangular_12.rs
use std::f64;
use std::cmp;

pub fn solve() {
    let mut i: usize = 2;
    let mut tri_num: usize = 1;
    let mut max: usize = 0;
    let mut v: usize = 0;
    loop {
        v = num_factors(tri_num);
        if v > max {
            max = v;
            println!("{}", max);
            if max > 500 {
                break;
            }
        }
        i += 1;
        tri_num += i;
    }
    println!("{} has +500 factors", tri_num);
}

fn num_factors(n: usize) -> usize {
    let mut i: usize = 1;
    let mut factors: usize = 0;
    let max_factor: usize = f64::sqrt(n as f64) as usize;
    while i < max_factor {
        if n % i == 0 {
            factors += 2;
        }
        i += 1;
    }
    factors
}
